bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java

```diff
@@ -224,6 +227,19 @@ public void loadConf(URL confURL) throws ConfigurationException {
         }
     }
 
+    /**
+     * Load configurations from environment variables prefixed with BK_.
+     */
+    public void loadEnv() {
```

Comment:
I am not sure this is the best place to put this code.
This is the configuration for the client as well.
I suggest to put this block into Main.java? This way it is specific to the bookie bootstrap procedure.

Reply:
I changed the loadConfEnv method of Main.java to the following

```
    private static void loadConfEnv(ServerConfiguration conf) throws IllegalArgumentException {
        try {
            System.getenv().entrySet().stream().filter(entry -> entry.getKey().startsWith(ENV_PREFIX))
                .forEach(entry -> {
                    String name = entry.getKey().substring(ENV_PREFIX.length());
                    String value = entry.getValue();
                    log.info("load environment {}={}", name, value);
                    conf.setProperty(name, value);
                });
            conf.validate();
        } catch (ConfigurationException e) {
            log.error("Malformed configuration file: {}", e);
            throw new IllegalArgumentException();
        }
    }
```

Do you have any suggestions about defining the location of constant ENV_PREFIX?

```
    // Environment Parameters Prefix
    protected static final String ENV_PREFIX = "BK_";
```

Reply:
> I am not sure this is the best place to put this code.
> This is the configuration for the client as well.
> I suggest to put this block into Main.java? This way it is specific to the bookie bootstrap procedure.

If this block is put into Main.java, it cannot be reused with LocalBookKeeper :(